serenity-screenplay-playwright/src/main/java/net/serenitybdd/screenplay/playwright/abilities/BrowseTheWebWithPlaywright.java

```diff
@@ -282,27 +324,14 @@ public void testFinishes(TestLifecycleEvents.TestFinished testFinished) {
             );
         }
         if (playwright != null) {
-
-            // Take screenshot for failed test
             BaseStepListener baseStepListener = StepEventBus.getEventBus().getBaseStepListener();
-            if (baseStepListener.currentTestFailed()) {
-                Page currentPage = getCurrentPage();
-                byte[] screenshot = currentPage.screenshot(new Page.ScreenshotOptions().setFullPage(true));
-
-                try {
-                    Path outputDirectory = baseStepListener.getOutputDirectory().toPath();
-                    Path pageSourceFile = Files.createTempFile(outputDirectory, "pagesource", ".txt");
-                    Path screenshotFile = Files.createTempFile(outputDirectory, "screenshot", ".png");
-                    Files.write(pageSourceFile, currentPage.content().getBytes(StandardCharsets.UTF_8));
-                    Files.write(screenshotFile, screenshot);
-
-                    ScreenshotAndHtmlSource screenshotAndHtmlSource = new ScreenshotAndHtmlSource(screenshotFile.toFile(), pageSourceFile.toFile());
-                    baseStepListener.firstFailingStep().ifPresent(
-                            step -> step.addScreenshot(screenshotAndHtmlSource)
-                    );
-                } catch (IOException e) {
-                    Assertions.fail("Failed to take Playwright screenshot", e);
-                }
+            // Take screenshot for failed test when SERENITY_TAKE_SCREENSHOTS is FOR_FAILURES
+            if (baseStepListener.currentTestFailed() && "FOR_FAILURES".equals(SERENITY_TAKE_SCREENSHOTS.from(environmentVariables))) {
```

Comment:
Rather than hard-coding the screenshot level, it would be better to use the `ScreenshotPermission` class, e.g.

```
        ScreenshotPermission screenshots = new ScreenshotPermission(configuration);
        if (screenshots.areAllowed(TakeScreenshots.FOR_FAILURES)) {...}
```

Reply:
Fix was applied and tested successfully. Thanks for the suggestion!